docs/flow/modin/core/dataframe/pandas/partitioning/axis_partition.rst

```diff
@@ -0,0 +1,44 @@
+BaseDataframeAxisPartition
+""""""""""""""""""""""""""
+
+The class is base for any axis partition class and serves as the last level on which
+operations that were conveyed from the partition manager are being performed on an entire column or row.
+
+The class provides an API that has to be overridden by the child classes in order to manipulate
+on a list of block partitions (making up column or row partition) they store.
+
+The procedures that use this class and its methods assume that they have some global knowledge
+about the entire axis. This may require the implementation to use concatenation or append on the
+list of block partitions.
+
+The ``PandasDataframeAxisPartition`` object that controls these objects (through the API exposed here) has an invariant
+that requires that this object is never returned from a function. It assumes that there will always be
+``PandasDataframeAxisPartition`` object stored and structures itself accordingly.
+
+.. warning::
+  The location of the ``BaseDataframeAxisPartition`` class in the `pandas` implementation of Modin Dataframe objects is a legacy.
```

Comment:
This will probably be moved in base as part of #3606.

Reply:
moved here as well